BlazarBase/src/main/java/com/hubspot/blazar/base/BranchSetting.java

```diff
@@ -0,0 +1,36 @@
+package com.hubspot.blazar.base;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class BranchSetting {
+
+  private final long branchId;
+  private final boolean triggerInterProjectBuilds;
+  private final boolean interProjectBuildOptIn;
+
+  @JsonCreator
+  public BranchSetting(@JsonProperty("branchId") long branchId,
+                       @JsonProperty("triggerInterProjectBuilds") boolean triggerInterProjectBuilds,
+                       @JsonProperty("interProjectBuildOptIn") boolean interProjectBuildOptIn) {
```

Comment:
@gchomatas would like some naming input here: 
`triggerInterProjectBuilds` is for auto starting inter project builds when this is triggered
`interProjectBuildOptIn` is for having this (non-master) branch get built as a part of inter project builds.

I've added the second one so that we can slim down the build trees a bit JavaUtils went from ~3300 items to ~650.
This tree now is just master and opted-in branches (hence the setting).

These names are a bit obtuse though, suggestions welcome.